helix-core/src/test/java/org/apache/helix/manager/zk/TestRawZkClient.java

```diff
@@ -745,4 +749,65 @@ public void testWaitForEstablishedSession() {
     // Recover zk server for later tests.
     _zkServer.start();
   }
+
+  @Test
+  public void testAsyncWriteOperations() {
+    ZkClient zkClient = new ZkClient(ZK_ADDR);
+    String originSizeLimit =
+        System.getProperty(ZkSystemPropertyKeys.ZK_SERIALIZER_ZNRECORD_WRITE_SIZE_LIMIT_BYTES);
+    System.setProperty(ZkSystemPropertyKeys.ZK_SERIALIZER_ZNRECORD_WRITE_SIZE_LIMIT_BYTES, "2000");
+    try {
+      zkClient.setZkSerializer(new ZNRecordSerializer());
+
+      ZNRecord oversizeZNRecord = new ZNRecord("Oversize");
+      StringBuilder sb = new StringBuilder(1204);
+      Random ran = new Random();
+      for (int i = 0; i < 1024; i++) {
+        sb.append(ran.nextInt(26) + 'a');
```

Comment:
Random is good to make sure compressed data is not that small!

Reply:
@jiajunwang `sb.append(ran.nextInt(26) + 'a');` doesn't add a char to the string builder but an integer. 
We still have to cast it: `sb.append((char) (ran.nextInt(26) + 'a'));`